database/factories/ModelFactory.php

```diff
@@ -26,6 +24,9 @@
     $url = app(AWS::class)->storeImage($upload, $faker->unique()->randomNumber(5));
 
     return [
+        'signup_id' => function () {
+            return factory(Signup::class)->create()->id;
+        },
         'url' => $url,
```

Comment:
do we need to associate the signup and post here or is this just so it doesn't get mad that posts are created without signups?

Reply:
Yep, this is because we were causing (and ignoring) foreign key constraints when we used to create posts w/o signups in the test suite. This makes sure that we always create and associate a corresponding signup.

Reply:
Looking at this again and now I'm confused about how the signup and post get associated?

Reply:
They get associated by setting `signup_id` on the post (since it's a "has many" relationship, which is defined on the `signup_id` column in the posts table).

Reply:
Here's some more [examples](https://laravel.com/docs/5.5/database-testing#relationships) from the Laravel docs. The nice thing about doing it this way is we don't have to remember to manually build the relationship in each test case.

Reply:
oh nice! You can just set the column! amazing!